core/storage/improvements/gae_models.py

```diff
@@ -0,0 +1,194 @@
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # Refers to the first entity version (inclusive) a task entry is relevant
+    # to.
+    entity_version_start = ndb.IntegerProperty(
+        required=True, indexed=True)
+    # Refers to the last entity version (exclusive) a task entry is relevant to.
+    entity_version_end = ndb.IntegerProperty(
+        default=None, required=False, indexed=True)
+    # ID of the user who closed the task, if any.
+    closed_by = ndb.StringProperty(
+        default=None, required=False, indexed=True)
+    # The date and time at which a task was closed or deprecated.
+    closed_on = ndb.DateTimeProperty(
+        default=None, required=False, indexed=True)
+    # Auto-generated string which provides a one-line summary of the task.
+    task_summary = ndb.StringProperty(
+        default=None, required=False, indexed=False)
+
+    @staticmethod
+    def get_user_id_migration_policy():
+        """TaskEntryModel has the closed_by field which refers to a user ID."""
+        return base_models.USER_ID_MIGRATION_POLICY.ONE_FIELD
+
+    @classmethod
+    def get_user_id_migration_field(cls):
+        """Return field that contains user ID."""
+        return cls.closed_by
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether any TaskEntryModel references the given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.get_user_id_migration_field() == user_id).iter().has_next()
+
+    @staticmethod
+    def get_deletion_policy():
+        """OK to delete task entries since they're just a historical record."""
+        return base_models.DELETION_POLICY.DELETE
+
+    @classmethod
+    def apply_deletion_policy(cls, user_id):
+        """Delete instances of TaskEntryModel for the user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be deleted.
+        """
+        cls.delete_multi(
+            cls.query(cls.get_user_id_migration_field() == user_id))
+
+    @staticmethod
+    def get_export_policy():
+        """TaskEntryModel contains the user ID that acted on a task."""
+        return base_models.EXPORT_POLICY.CONTAINS_USER_DATA
+
+    @staticmethod
+    def export_data(user_id):
+        """Returns the user-relevant properties of TaskEntryModels.
+
+        Args:
+            user_id: str. The ID of the user whose data should be exported.
+
+        Returns:
+            dict. The user-relevant properties of TaskEntryModel in a dict
+            format. In this case, we are returning all the ids of the tasks
+            which were closed by this user.
+        """
+        tasks_closed_by_user = TaskEntryModel.query(
+            TaskEntryModel.get_user_id_migration_field() == user_id)
```

Comment:
Please use the field here directly since `TaskEntryModel.get_user_id_migration_field` will be removed after the migration is completed.